httpserver/src/main/kotlin/tbdex/sdk/httpserver/handlers/SubmitOrder.kt

```diff
@@ -0,0 +1,89 @@
+package tbdex.sdk.httpserver.handlers
+
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.request.receiveText
+import io.ktor.server.response.respond
+import tbdex.sdk.httpclient.models.ErrorDetail
+import tbdex.sdk.httpserver.models.CallbackError
+import tbdex.sdk.httpserver.models.ErrorResponse
+import tbdex.sdk.httpserver.models.ExchangesApi
+import tbdex.sdk.httpserver.models.SubmitCallback
+import tbdex.sdk.protocol.models.Message
+import tbdex.sdk.protocol.models.MessageKind
+import tbdex.sdk.protocol.models.Order
+import tbdex.sdk.protocol.models.Quote
+
+/**
+ * Handles the submission of an order by parsing the incoming message, validating the submission,
+ * and executing the specified callback if provided.
+ *
+ * @param call The [ApplicationCall] instance representing the HTTP call.
+ * @param exchangesApi The [ExchangesApi] instance for interacting with TBDex exchanges.
+ * @param callback The optional callback function to be executed after successful order submission.
+ */
+@Suppress("TooGenericExceptionCaught", "MaxLineLength", "SwallowedException")
+suspend fun submitOrder(
+  call: ApplicationCall,
+  exchangesApi: ExchangesApi,
+  callback: SubmitCallback?
+) {
+  val message: Order?
+
+  try {
+    message = Message.parse(call.receiveText()) as Order
```

Comment:
Same as above:
Kotlin question: Is this as Order safe to do? It looks like we verify the message is a valid message of some kind, but I'm not sure if we can be confident that it's actually a Order.

Reply:
see response above